### PR TITLE
printf: Default left-justify integer conversion to 1 width

### PR DESCRIPTION
**Pull request by hbina**
When using left-justify with integer conversion (like `printf '%-o'`),
default the minimum width to 1.

Closes: https://github.com/uutils/coreutils/issues/3050

Signed-off-by: Hanif Ariffin <hanif.ariffin.4326@gmail.com>

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
The logic of the `SubParser` code is a bit hard to follow. Do you know if there is a possibility where `min_width_tmp` is not just empty but an invalid integer? Is that a case that we have to account for?

**Comment by sylvestre** [comment]
@hbina ping?

**Comment by hbina** [comment]
I don't fully understand the [Output Conversion Syntax](https://www.gnu.org/software/libc/manual/html_node/Output-Conversion-Syntax.html) but here's a couple of examples of this at work:

```shell
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> cargo run --quiet -- printf '%-10o' 3
3         ⏎                                                                                                                                                                                             
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> cargo run --quiet -- printf '%10o' 3
         3⏎                                                                                                                                                                                                
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> printf '%-10o' 3
3         ⏎                                                                                                                                                                                                
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> printf '%10o' 3
         3⏎                                                                                                                                                                                                
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> cargo run --quiet -- printf '%-10o'
0         ⏎                                                                                                                                                                                                
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> cargo run --quiet -- printf '%10o'
         0⏎                                                                                                                                                                                             
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> printf '%-10o'
0         ⏎                                                                                                                                                                                                
hbina@akarin ~/g/uutils (hbina-printf-default-left-justify-legth)> printf '%10o'
         0⏎                                                                                    
```



>Do you know if there is a possibility where min_width_tmp is not just empty but an invalid integer? Is that a case that we have to account for?
GNU says that if the conversion syntax is ill-formed, the behavior is simply unpredictable so I don't think we have to care about invalid input

```
If the syntax of a conversion specification is invalid, unpredictable things will happen, so don’t do this. If there aren’t enough function arguments provided to supply values for all the conversion specifications in the template string, or if the arguments are not of the correct types, the results are unpredictable. If you supply more arguments than conversion specifications, the extra argument values are simply ignored; this is sometimes useful.
